hcl2template/types.build.hcp_packer_registry.go

```diff
@@ -54,7 +54,7 @@ func (p *Parser) decodeHCPRegistry(block *hcl.Block, cfg *PackerConfig) (*HCPPac
 		return nil, diags
 	}
 
-	if !bucketNameRegexp.MatchString(b.Slug) {
+	if b.Slug != "" && !bucketNameRegexp.MatchString(b.Slug) {
```

Comment:
May be worth adding a comment explaining that if bucket is unset its fine because it can be set in an environment variable

Reply:
Yeah that can't hurt; I'll check that while I figure out the cause for the crash during testing